src/main/java/org/datadog/jmxfetch/tasks/TaskProcessor.java

```diff
@@ -38,7 +38,7 @@ public void setThreadPoolExecutor(ExecutorService executor) {
      * */
     public boolean ready() {
         ThreadPoolExecutor tpe = (ThreadPoolExecutor) threadPoolExecutor;
-        return !(tpe.getPoolSize() == tpe.getActiveCount());
+        return !(tpe.getMaximumPoolSize() == tpe.getActiveCount());
```

Comment:
I think for a fixed thread pool these two are equivalent, but this is better and less error-prone as we also had some cached thread-pool support (although you've killed it).

Reply:
should've mentioned this earlier, but for a fixed thread pool this isn't equivalent: before the first JMXFetch "iteration", `tpe.getPoolSize()` is `0`, so the app considers the pool as not ready and prints out warnings before the first collection is actually kicked off.

Reply:
Interesting. Nice catch.